projects/wave-core-new/src/lib/datasets/dataset/dataset.component.ts

```diff
@@ -0,0 +1,72 @@
+import {Component, OnInit, ChangeDetectionStrategy, Input} from '@angular/core';
+import {DataSet, getDataSetType} from '../dataset.model';
+import {RasterLayer, VectorLayer} from '../../layers/layer.model';
+import {MappingRasterSymbology, PointSymbology} from '../../layers/symbology/symbology.model';
+import {Unit} from '../../operators/unit.model';
+import {ProjectService} from '../../project/project.service';
+import {mergeMap} from 'rxjs/operators';
+
+@Component({
+    selector: 'wave-dataset',
+    templateUrl: './dataset.component.html',
+    styleUrls: ['./dataset.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DataSetComponent implements OnInit {
+
+    @Input() dataset: DataSet;
+
+    constructor(private projectService: ProjectService) {
+    }
+
+    ngOnInit(): void {
+    }
+
+    add() {
+        const workflow = {
+            type: getDataSetType(this.dataset),
+            operator: {
+                type: this.dataset.source_operator,
+                params: {
+                    data_set: this.dataset.id,
+                },
+            }
+        };
+
+        this.projectService.registerWorkflow(workflow).pipe(
+            mergeMap(workflowId => {
+                if (getDataSetType(this.dataset) === 'Raster') {
+                    return this.projectService.addLayer(new RasterLayer({
+                        workflowId,
+                        name: this.dataset.name,
+                        symbology: new MappingRasterSymbology({
+                            opacity: 1,
+                            // TODO: insert proper unit
```

Comment:
You could get measurement and unit by calling the metadata of the workflow.

Reply:
true, it's part of the `RasterResultDescriptor`. However
- the measurement doesn't have min/max values yet
- there is no raster source that works with the new DataSets so I can't test ist properly
That's why i would like to postpone this